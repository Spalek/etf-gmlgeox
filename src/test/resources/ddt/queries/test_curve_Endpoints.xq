import module namespace ggeo = 'https://modules.etf-validator.net/gmlgeox/2';

declare namespace gml = 'http://www.opengis.net/gml/3.2';
declare namespace ii = 'http://www.interactive-instruments.de/test';

declare variable $dbname external := 'GMLGEOX-JUNIT-TEST-DB-000';
let $members := db:open($dbname)//ii:member
let $init := ggeo:init($dbname)

let $geometries := $members/*
return
 <test_curveEndPoints>    
   {
    for $geom in $geometries
    order by $geom/@gml:id
    let $endPoints := ggeo:curveEndPoints($geom)    
    let $result := string-join(($endPoints ! ggeo:toWKT(.)),', ')
    return 
      <test geom='{$geom/@gml:id}'>{$result}</test>    
   }
 </test_curveEndPoints>
