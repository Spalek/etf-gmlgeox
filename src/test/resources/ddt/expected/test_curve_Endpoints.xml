<test_curveEndPoints>
  <test geom="c1">POINT (1 1.5), POINT (2 2.5)</test>
  <test geom="c2">POINT (3 3), POINT (5 5)</test>
  <test geom="c3">POINT (6 1), POINT (8 1)</test>
  <test geom="c4">POINT (10 10), POINT (10 10)</test>
  <test geom="p1"/>
  <test geom="s1"/>
</test_curveEndPoints>